wideq/core.py

```diff
@@ -20,6 +21,50 @@
 DATE_FORMAT = '%a, %d %b %Y %H:%M:%S +0000'
 
 
+def setup_logging() -> None:
+    level = logging.INFO
+    fmt = "%(asctime)s %(levelname)s [%(name)s] %(message)s"
+    datefmt = "%Y-%m-%d %H:%M:%S"
+    try:
+        from colorlog import ColoredFormatter  # type: ignore
+        # basicConfig must be called after importing colorlog in order to
+        # ensure that the handlers it sets up wraps the correct streams.
+        logging.basicConfig(level=level)
+
+        colorfmt = f"%(log_color)s{fmt}%(reset)s"
+        logging.getLogger().handlers[0].setFormatter(
+            ColoredFormatter(
+                colorfmt,
+                datefmt=datefmt,
+                reset=True,
+                log_colors={
+                    "DEBUG": "cyan",
+                    "INFO": "green",
+                    "WARNING": "yellow",
+                    "ERROR": "red",
+                    "CRITICAL": "red",
+                },
+            )
+        )
+    except ImportError:
+        pass
+
+    logging.basicConfig(format=fmt, datefmt=datefmt, level=logging.INFO)
+
+    logging.getLogger("urllib3").setLevel(logging.WARNING)
+    logging.getLogger("requests").setLevel(logging.WARNING)
+    logger = logging.getLogger("")
```

Comment:
Maybe we should create a “wideq” logger instead of using the root one? This could help avoid bad interactions with other modules that also use logging.

Reply:
Actually, I completely agree, especially when the code is run inside home assistant and other projects that makes a lot more sense, good point. I was still working with mostly the example in mind.